aries_cloudagent/core/plugin_registry.py

```diff
@@ -195,9 +195,13 @@ def register_package(self, package_name: str) -> Sequence[ModuleType]:
 
     async def init_context(self, context: InjectionContext):
         """Call plugin setup methods on the current context."""
-        for plugin in self._plugins.values():
+        plugins_config = context.settings.get("plugins_config", {})
+        context.update_settings({"plugins_config": None})
+        for key, plugin in self._plugins.items():
             if hasattr(plugin, "setup"):
+                context.update_settings({"plugin_config": plugins_config.get(key)})
                 await plugin.setup(context)
+                context.update_settings({"plugin_config": None})
```

Comment:
Sorry to continue to pester about this chunk of code; could you explain why it's necessary to isolate the individual plugin configurations?

Reply:
hahaha, no problem. I thought that some config could be access tokens, passwords, mails, whatever... So I don't think It could be a good idea to share the configuration with other pieces of code (plugins) that could have security breaches or be malware (?).

For instance, for the Kafka implementation, we should have a user/password, broker IPs...

Reply:
I see your point, thanks for clarifying. As we discussed earlier, I don't think this would prevent malicious code from doing bad things. For instance, a malicious plugin could just read the plugin config from file again. At this stage in ACA-Py's development, plugins must be considered trusted. If the plugin is from a third party, developers are responsible for evaluating the plugin for disagreeable behavior. Since we don't have a good answer for how we enforce boundaries between plugins or for executing untrusted plugins, I think we should simply keep it simple and allow the plugin configuration to be read by anything with access to the context. If there are values that are truly sensitive, I would argue that the correct action there is to not use the plugin configuration and instead let the plugin worry about how to securely obtain those values.